src/Symfony/Component/Console/Tests/Command/InvokableCommandTest.php

```diff
@@ -277,22 +279,45 @@ public function testNonBinaryInputOptions(array $parameters, array $expected)
 
     public static function provideNonBinaryInputOptions(): \Generator
     {
-        yield 'defaults' => [[], [null, 'b', []]];
-        yield 'with-value' => [['--a' => 'x', '--b' => 'y', '--c' => ['z']], ['x', 'y', ['z']]];
-        yield 'without-value' => [['--a' => null, '--b' => null, '--c' => null], [null, null, null]];
+        yield 'defaults' => [[], ['', '', [], ['a', 'b']]];
+        yield 'with-value' => [['--a' => 'x', '--b' => 'y', '--c' => ['z'], '--d' => ['c', 'd']], ['x', 'y', ['z'], ['c', 'd']]];
+        yield 'without-value' => [['--b' => null], ['', null, [], ['a', 'b']]];
     }
 
-    public function testInvalidOptionDefinition()
+    /**
+     * @dataProvider provideInvalidOptionDefinitions
+     */
+    public function testInvalidOptionDefinition(callable $code, string $expectedMessage)
     {
         $command = new Command('foo');
-        $command->setCode(function (#[Option] string $a) {});
+        $command->setCode($code);
 
         $this->expectException(LogicException::class);
-        $this->expectExceptionMessage('The option parameter "$a" must declare a default value.');
+        $this->expectExceptionMessage($expectedMessage);
 
         $command->getDefinition();
     }
 
+    public static function provideInvalidOptionDefinitions(): \Generator
+    {
+        yield 'no-default' => [
+            function (#[Option] string $a) {},
+            'The option parameter "$a" must declare a default value.',
+        ];
+        yield 'nullable-bool-default-true' => [
+            function (#[Option] ?bool $a = true) {},
+            'The option parameter "$a" must not be nullable when it has a default boolean value.',
+        ];
+        yield 'nullable-bool-default-false' => [
+            function (#[Option] ?bool $a = false) {},
+            'The option parameter "$a" must not be nullable when it has a default boolean value.',
+        ];
+        yield 'nullable-string' => [
+            function (#[Option] ?string $a = null) {},
+            'The option parameter "$a" must not have a default of null.',
+        ];
```

Comment:
can we add the exception for `?array` too?

Reply:
Done.